SYCL/Config/check_config_processing.cpp

```diff
@@ -0,0 +1,128 @@
+    return 0;
+  }
+  if (envir == "AFTER_ASSIGNMENT") {
+    std::cout << "a= b" << std::endl;
+    return 0;
+  }
+  if (envir == "BIG_NAME") {
+    // Max variable name is 256 characters
+    std::cout
+        << "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+           "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+           "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+           "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa=b"
+        << std::endl;
+    return 0;
+  }
+  if (envir == "SMALL_NAME") {
+    std::cout << "=b" << std::endl;
+    return 0;
+  }
+  if (envir == "BIG_VALUE") {
+    // Max variable value contains 256 or 1024 characters
+    std::cout
+        << "a="
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+           "bbbbbbbbbbbbbbbbbbbb"
+        << std::endl;
+    return 0;
+  }
+  if (envir == "SMALL_VALUE") {
+    std::cout << "a=" << std::endl;
+    return 0;
+  }
+  if (envir == "HARD") {
+    // To check that the processing reaches "a="
+    std::cout << "#\n"
+              << "  #\n"
+              << "#a=b"
+              << "\n\n\n"
+              << "a\n"
+              << " aaa \n"
+              << "a=b\n"
+              << "a=\n"
+              << std::endl;
+    return 0;
+  }
+
+  if (envir == "CHECK") {
+    try {
+      auto device = sycl::device{sycl::default_selector{}};
+    } catch (sycl::exception &e) {
+      std::string errorMessage = e.what();
+      if ((errorMessage == "SPACE found at the beginning/end of the line or "
+                           "before/after '='") ||
+          (errorMessage ==
+           "Variable name is more than 256 or less than one character") ||
+          (errorMessage == "The value contains more than 256 characters or "
```

Comment:
Need to skip comparison with the specific number during checking of error message (e.g., using regex)